File: nhom12/src/main/java/nhom12/example/nhom12/model/Product.java
package nhom12.example.nhom12.model;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.experimental.SuperBuilder;
import org.springframework.data.mongodb.core.index.Indexed;
import org.springframework.data.mongodb.core.mapping.Document;

@Document(collection = "products")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@SuperBuilder
public class Product extends BaseDocument {

  @Indexed private String name;

  private String brand;
  private String categoryId;
  private double price;
  private Double originalPrice;
  private String image;
  private double rating;
  private String badge;
  private String specs;
}

File: frontend/src/types/product.ts
export interface Product {
  id: string;
  name: string;
  brand: string;
  categoryId?: string;
  categoryName?: string;
  price: number;
  originalPrice?: number;
  image: string;
  rating: number;
  badge?: string;
  specs?: string;
  createdAt?: string;
  updatedAt?: string;
}

export interface CreateProductPayload {
  name: string;
  brand: string;
  categoryId?: string;
  price: number;
  originalPrice?: number;
  image: string;
  rating?: number;
  badge?: string;
  specs?: string;
}

export interface Category {
  id: string;
  name: string;
  slug: string;
  description?: string;
  icon?: string;
  createdAt?: string;
}

export interface CreateCategoryPayload {
  name: string;
  description?: string;
  icon?: string;
}
